internal/services/compute/managed_disk_resource.go

```diff
@@ -356,6 +361,21 @@ func resourceManagedDiskCreate(d *pluginsdk.ResourceData, meta interface{}) erro
 		}
 	}
 
+	if d.Get("bursting_enabled").(bool) {
+		switch storageAccountType {
+		case string(compute.StorageAccountTypesPremiumLRS):
+		case string(compute.StorageAccountTypesPremiumZRS):
+		default:
+			return fmt.Errorf("`bursting_enabled` can only be set to true when `storage_account_type` is set to `Premium_LRS` or `Premium_ZRS`")
+		}
+
+		if diskSizeGB != 0 && diskSizeGB <= 512 {
```

Comment:
the API appears to also support this for smaller Standard SSD's too:

> An on-demand bursting model, where the disk bursts whenever its needs exceed its current capacity. This model incurs additional charges anytime the disk bursts. On-demand bursting is only available for premium SSDs larger than 512 GiB.
> A credit-based model, where the disk will burst only if it has burst credits accumulated in its credit bucket. This model does not incur additional charges when the disk bursts. Credit-based bursting is only available for premium and standard SSDs 512 GiB and smaller.

https://docs.microsoft.com/en-us/azure/virtual-machines/disk-bursting#disk-level-bursting

As such I think this check could ultimately be removed?

Reply:
The credit-based bursting is enabled by default on all eligible disks, including Standard and Premium. The property here only controls on-demand bursting. But seems like the name is confusing, I'm changing the property name to `on_demand_bursting_enabled`.
https://docs.microsoft.com/en-us/azure/virtual-machines/disks-enable-bursting
![image](https://user-images.githubusercontent.com/10579712/141427085-655707c9-f899-4490-8266-ef6b03c33054.png)